src/otaclient_common/linux.py

```diff
@@ -264,31 +265,28 @@ def subprocess_run_wrapper(
     if isinstance(cmd, str):
         cmd = shlex.split(cmd)
 
-    preexec_fn: Optional[Callable[..., Any]] = None
-    if chroot or set_host_mnt_ns:
-
-        def _preexec():
-            if chroot:
-                os.chroot(chroot)
-                os.chdir("/")
-
-            # NOTE(20251029): only support sent back to host mnt ns
-            if set_host_mnt_ns:
-                setns_wrapper(_root_mnt_ns)
-
-        preexec_fn = _preexec
```

Comment:
Using preexec_fn is still not so robust, as preexec_fn will be executed after forking before execv, which means the current python runtime will be also used in the newly forked process. Python documentation indicates there is potential risk of doing so when the process is multi-threaded.

This time we change to use `chroot` and `nsenter` commands to achieve the goal, while reducing risk and improve compatibility and robustness. 
`chroot` is included in `coreutils` package, while `nsenter` is included in `util-linux` package, both are core components of modern linux distro and installed by default.